### PR TITLE
SNMP Community - Symbols Cause Churn

### PR DESCRIPTION
**Pull request by mikewiebe**
**Summary:**
This update fixes the issue with the `cisco_snmp_community` provider described in https://github.com/cisco/cisco-network-puppet-module/issues/414

This is a two part fix:
 - Refactor the underlying `snmp_community` object to use nxapi structured output instead of raw cli based output.
     - NOTE: Structured output for this provider is currently only supported on `n3k`, `n9k` and `n9k-f` platforms.  `n5k`, `n6k` and `n7k` will continue to use raw cli based nxapi and hence with this fix will still be unable to handle complex `cisco_snmp_community` names.  A future commit will address this for raw cli based output.
- The infrastructure handling nxapi structured output was refactored to better query data that may include special complex characters.

Tested on: `n3k`, `n9k`, `n9k-f`, `n5k`, `n6k`, `n7k`

### CODE REVIEW

File: lib/cisco_node_utils/node.rb [hunk 1]
```diff
@@ -87,8 +87,9 @@ def massage_structured(get_args, ref)
         ref.hash['get_value'] = get_args[:value]
         ref.hash['drill_down'] = true
         get_args[:value] = nil
+        cache_flush
       end
-      cache_flush
+      # cache_flush
```
Comment (saichint) [inline lib/cisco_node_utils/node.rb]: Remove this line

**Comment by saichint** [comment]
Only a minor comment removal.
Also take a look at this:
https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/node.rb#L83
I think we don't need to check the condition "if ref.hash['get_value'].is_a?(Array)" as the structured output is an array already. Add a comment to remove this condition in the future.

**Comment by mikewiebe** [comment]
@saichint Addressed your PR comments.

File: lib/cisco_node_utils/client/utils.rb [hunk 2]
```diff
@@ -130,7 +130,7 @@ def self.to_regexp(input)
   # @param keys [Array] lookup sequence
   def self.filter_data(data: nil,
                        keys: nil)
-    return nil if data.nil?
+    return nil if data.nil? || data.empty?
```
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/client/utils.rb:133]: note for next time...
You can combine these into one check:

```
return nil if data.to_s.empty?
```

**Pull request merged**